Adobe provider screen views with props

arifken opened the pull request:
In our Omniture implementation, all screen views have custom properties associated with them. This PR overrides `-didShowNewPageView:withProperties:` to call `-trackState:data:` for page views that have custom context data. Without this PR, ARAnalytics triggers an event (`-trackAction:data:`). Is this something that would be useful to other developers? or is this too specific to our implementation? Thanks

orta commented:
> Is this something that would be useful to other developers? or is this too specific to our implementation? Thanks

@synthemesc - do you have an opinion on this? I don't really know enough about the Adobe SDK to make a call. Otherwise it makes sense to merge

synthemesc commented:
It would probably break a lot of existing "workflow" with our analytics to do this by default. @arifken, any way you can expose a setting and default to the non-tracking behavior?

arifken commented:
@synthemesc yes definitely, good call. Will do and I'll update the PR

arifken commented:
Updated :) I also added a unit test for the AdobeProvider, and increased the version of OCMock to take advantage of new syntax features. Let me know if you think this is ok. Thanks

synthemesc commented:
LGTM. +1

orta commented:
Ace, @arielelkin - if you can rebase and ping me, I'll happily merge this :+1:

arielelkin commented:
@orta you probably mean @arifken :-)

orta commented:
doh, thanks!

arifken commented:
rebased! thanks @orta!
